File: programmers/[1차] 비밀지도/solution.cpp
#include <string>
#include <vector>
#include <bitset>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2)
{
    vector<string> answer;

    for (int i = 0; i < n; i++)
    {
        bitset<16> line1(arr1[i]);
        bitset<16> line2(arr2[i]);

        string s1 = line1.to_string();
        string s2 = line2.to_string();
        string temp = "";

        for (int j = 16 - n; j < 16; j++)
        {
            if (s1[j] == '1' || s2[j] == '1')
                temp += "#";
            else
                temp += " ";
        }

        answer.push_back(temp);
    }

    return answer;
}

File: programmers/[1차] 비밀지도/solution.py
def solution(n, arr1, arr2):
    answer = []

    for i in range(n):
        b1 = format(arr1[i], 'b')
        if len(b1) != n:
            while len(b1) != n:
                b1 = "0" + b1

        b2 = format(arr2[i], 'b')
        if len(b2) != n:
            while len(b2) != n:
                b2 = "0" + b2

        s = ""
        for j in range(n):
            if b1[j] == "1" or b2[j] == "1":
                s += "#"
            else:
                s += " "

        answer.append(s)

    return answer
